### ISSUE TITLE
[Bug]: [benchmark][cluster]During the stability test, datacoord's CPU and memory kept going up, not down.

### ISSUE THREAD

**Issue by wangting0128** [milvus-io/milvus#16394]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20220402-028c5cb8
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): 2.0.2.dev9
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

argo task: benchmark-tag-b8htt

test yaml:
client-configmap:client-random-locust-100m-ddl-r8-w2
server-configmap:server-cluster-8c64m

client pod: benchmark-tag-b8htt-3255216514

server:
```
NAME                                                              READY   STATUS      RESTARTS   AGE     IP             NODE                      NOMINATED NODE   READINESS GATES
benchmark-tag-b8htt-1-etcd-0                                      1/1     Running     0          3d15h   10.97.17.8     qa-node014.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-etcd-1                                      1/1     Running     0          3d15h   10.97.16.17    qa-node013.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-etcd-2                                      1/1     Running     0          3d15h   10.97.17.9     qa-node014.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-milvus-datacoord-58c78c8787-54t9b           1/1     Running     0          3d15h   10.97.11.77    qa-node009.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-milvus-datanode-6f8cdfcc4b-ztldq            1/1     Running     0          3d15h   10.97.20.117   qa-node018.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-milvus-indexcoord-644bd47d98-w84r6          1/1     Running     0          3d15h   10.97.11.79    qa-node009.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-milvus-indexnode-7564996c47-pdwpm           1/1     Running     0          3d15h   10.97.20.119   qa-node018.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-milvus-proxy-5bc7cd94d4-r69zr               1/1     Running     0          3d15h   10.97.11.85    qa-node009.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-milvus-querycoord-5b475b4f5c-f88h9          1/1     Running     0          3d15h   10.97.11.78    qa-node009.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-milvus-querynode-6cbc595865-njbwf           1/1     Running     0          3d15h   10.97.20.118   qa-node018.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-milvus-rootcoord-76d56d8f95-7kb8p           1/1     Running     0          3d15h   10.97.11.76    qa-node009.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-minio-0                                     1/1     Running     0          3d15h   10.97.20.121   qa-node018.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-minio-1                                     1/1     Running     0          3d15h   10.97.19.135   qa-node016.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-minio-2                                     1/1     Running     0          3d15h   10.97.19.132   qa-node016.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-minio-3                                     1/1     Running     0          3d15h   10.97.19.136   qa-node016.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-pulsar-bookie-0                             1/1     Running     0          3d15h   10.97.19.131   qa-node016.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-pulsar-bookie-1                             1/1     Running     0          3d15h   10.97.17.5     qa-node014.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-pulsar-bookie-2                             1/1     Running     0          3d15h   10.97.5.48     qa-node003.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-pulsar-bookie-init-wbg4p                    0/1     Completed   0          3d15h   10.97.18.4     qa-node017.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-pulsar-broker-0                             1/1     Running     0          3d15h   10.97.12.238   qa-node015.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-pulsar-proxy-0                              1/1     Running     0          3d15h   10.97.3.22     qa-node001.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-pulsar-pulsar-init-gkwq8                    0/1     Completed   0          3d15h   10.97.4.252    qa-node002.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-pulsar-recovery-0                           1/1     Running     0          3d15h   10.97.16.15    qa-node013.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-pulsar-zookeeper-0                          1/1     Running     0          3d15h   10.97.4.253    qa-node002.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-pulsar-zookeeper-1                          1/1     Running     0          3d15h   10.97.18.5     qa-node017.zilliz.local   <none>           <none>
benchmark-tag-b8htt-1-pulsar-zookeeper-2                          1/1     Running     0          3d15h   10.97.9.73     qa-node007.zilliz.local   <none>           <none>
```

monitor:
<img width="1371" alt="截屏2022-04-06 11 47 41" src="https://user-images.githubusercontent.com/26307815/161891942-cc708ed6-654e-417b-a83e-f592e02ec5dc.png">

### Steps To Reproduce

```markdown
1、create collection
2、create index of ivf_sq8
3、insert 100m million vectors
4、flush collection
5、build index with the same params
6、load collection
7、locust concurrent: query<-search, load, get<-query, scene_test
```

### Anything else?

client-random-locust-100m-ddl-r8-w2:
```
locust_random_performance:
      collections:
        -
          collection_name: sift_100m_128_l2
          ni_per: 50000
          build_index: true
          index_type: ivf_sq8
          index_param:
            nlist: 2048
          task:
            types:
              -
                type: query
                weight: 8
                params:
                  top_k: 10
                  nq: 10
                  search_param:
                    nprobe: 16
              -
                type: load
                weight: 1
              -
                type: get
                weight: 8
                params:
                  ids_length: 10
              -
                type: scene_test
                weight: 2
            connection_num: 1
            clients_num: 20
            spawn_rate: 2
            during_time: 302400

```

**Comment by elstic** [comment]
During the stability test, datacoord's CPU and memory kept going up, not down.


### Current Behavior
argo task: fouram-tag-no-clean-8pt9c
server-instance
fouram-tag-no-clean-8pt9c-1
server-configmap
server-cluster-8c64m-querynode2
client-configmap
client-random-locust-search-filter-100m-ddl-replica2

server:
```
fouram-tag-no-clean-8pt9c-1-etcd-0                               1/1     Running     0              5m59s   10.104.5.233   4am-node12   <none>           <none>
fouram-tag-no-clean-8pt9c-1-etcd-1                               1/1     Running     0              5m59s   10.104.9.19    4am-node14   <none>           <none>
fouram-tag-no-clean-8pt9c-1-etcd-2                               1/1     Running     0              5m59s   10.104.4.131   4am-node11   <none>           <none>
fouram-tag-no-clean-8pt9c-1-milvus-datacoord-dd585665f-4ccgs     1/1     Running     1 (118s ago)   5m59s   10.104.4.122   4am-node11   <none>           <none>
fouram-tag-no-clean-8pt9c-1-milvus-datanode-5ccf545598-mjlzf     1/1     Running     1 (119s ago)   5m59s   10.104.9.16    4am-node14   <none>           <none>
fouram-tag-no-clean-8pt9c-1-milvus-indexcoord-844c9849db-n8xxs   1/1     Running     1 (118s ago)   5m59s   10.104.5.225   4am-node12   <none>           <none>
fouram-tag-no-clean-8pt9c-1-milvus-indexnode-5f77b58647-zznb9    1/1     Running     0              5m59s   10.104.1.67    4am-node10   <none>           <none>
fouram-tag-no-clean-8pt9c-1-milvus-proxy-6d744d96c8-2lgfd        1/1     Running     1 (117s ago)   5m59s   10.104.5.226   4am-node12   <none>           <none>
fouram-tag-no-clean-8pt9c-1-milvus-querycoord-65b4f4554f-8jb7z   1/1     Running     1 (118s ago)   5m59s   10.104.4.123   4am-node11   <none>           <none>
fouram-tag-no-clean-8pt9c-1-milvus-querynode-74bfd9f94f-6brp4    1/1     Running     0              5m59s   10.104.9.17    4am-node14   <none>           <none>
fouram-tag-no-clean-8pt9c-1-milvus-querynode-74bfd9f94f-q8lgg    1/1     Running     0              5m59s   10.104.5.229   4am-node12   <none>           <none>
fouram-tag-no-clean-8pt9c-1-milvus-rootcoord-55d655df-ctzm5      1/1     Running     1 (117s ago)   5m59s   10.104.5.227   4am-node12   <none>           <none>
fouram-tag-no-clean-8pt9c-1-minio-0                              1/1     Running     0              5m59s   10.104.5.235   4am-node12   <none>           <none>
fouram-tag-no-clean-8pt9c-1-minio-1                              1/1     Running     0              5m59s   10.104.4.127   4am-node11   <none>           <none>
fouram-tag-no-clean-8pt9c-1-minio-2                              1/1     Running     0              5m59s   10.104.1.71    4am-node10   <none>           <none>
fouram-tag-no-clean-8pt9c-1-minio-3                              1/1     Running     0              5m58s   10.104.9.21    4am-node14   <none>           <none>
fouram-tag-no-clean-8pt9c-1-pulsar-bookie-0                      1/1     Running     0              5m59s   10.104.4.130   4am-node11   <none>           <none>
fouram-tag-no-clean-8pt9c-1-pulsar-bookie-1                      1/1     Running     0              5m58s   10.104.1.72    4am-node10   <none>           <none>
fouram-tag-no-clean-8pt9c-1-pulsar-bookie-2                      1/1     Running     0              5m58s   10.104.5.238   4am-node12   <none>           <none>
fouram-tag-no-clean-8pt9c-1-pulsar-bookie-init-p97zh             0/1     Completed   0              5m59s   10.104.4.120   4am-node11   <none>           <none>
fouram-tag-no-clean-8pt9c-1-pulsar-broker-0                      1/1     Running     0              5m59s   10.104.4.124   4am-node11   <none>           <none>
fouram-tag-no-clean-8pt9c-1-pulsar-proxy-0                       1/1     Running     0              5m59s   10.104.4.121   4am-node11   <none>           <none>
fouram-tag-no-clean-8pt9c-1-pulsar-pulsar-init-qsmmb             0/1     Completed   0              5m59s   10.104.5.228   4am-node12   <none>           <none>
fouram-tag-no-clean-8pt9c-1-pulsar-recovery-0                    1/1     Running     0              5m59s   10.104.5.224   4am-node12   <none>           <none>
fouram-tag-no-clean-8pt9c-1-pulsar-zookeeper-0                   1/1     Running     0              5m59s   10.104.5.234   4am-node12   <none>           <none>
fouram-tag-no-clean-8pt9c-1-pulsar-zookeeper-1                   1/1     Running     0              5m12s   10.104.4.133   4am-node11   <none>           <none>
fouram-tag-no-clean-8pt9c-1-pulsar-zookeeper-2                   1/1     Running     0              4m33s   10.104.1.74    4am-node10   <none>           <none>
```




datacoord cpu usage: 

![image](https://user-images.githubusercontent.com/48523564/216039258-a1acf6a2-1a49-46d7-bf6a-93ef41f8ca41.png)
datacoord memory usage: 
![image](https://user-images.githubusercontent.com/48523564/216039624-1ba46cfd-42f4-4381-ba94-f7dca0ab3a7d.png)

### Steps To Reproduce
1、create   collection
2、create index of IVF_SQ8
3、insert 100m million vectors
4、flush collection
5、build index with the same params
6、load collection
7、locust concurrent:  search, load,  query, scene_test

**Comment by elstic** [comment]
@soothing-rain Please take a look at the question that.  and the fact that these two problems occur simultaneously: https://github.com/milvus-io/milvus/issues/21897

**Comment by soothing-rain** [comment]
> @soothing-rain Please take a look at the question that. and the fact that these two problems occur simultaneously: #21897

What do you mean by "occur simultaneously"? Did you spot that DataCoord's CPU & memory usage keeps growing infinitely?

**Comment by elstic** [comment]
> What do you mean by "occur simultaneously"? Did you spot that DataCoord's CPU & memory usage keeps growing infinitely?

It means that when we run stability tests, the case has a query, load, search latency increase phenomenon and datacoord CPU rise problem

**Comment by soothing-rain** [comment]
> > What do you mean by "occur simultaneously"? Did you spot that DataCoord's CPU & memory usage keeps growing infinitely?
> 
> It means that when we run stability tests, the case has a query, load, search latency increase phenomenon and datacoord CPU rise problem

I see.

@wangting0128 Do you think we can reproduce this test on 2.2.3 branch and see if there's DataCoord CPU rise?

**Comment by wangting0128** [comment]
> > > What do you mean by "occur simultaneously"? Did you spot that DataCoord's CPU & memory usage keeps growing infinitely?
> > 
> > 
> > It means that when we run stability tests, the case has a query, load, search latency increase phenomenon and datacoord CPU rise problem
> 
> I see.
> 
> @wangting0128 Do you think we can rerun this test on 2.2.3 branch and see if there's DataCoord CPU rise?

@elstic Please follow up on this question, thanks

**Comment by elstic** [comment]
> > > > What do you mean by "occur simultaneously"? Did you spot that DataCoord's CPU & memory usage keeps growing infinitely?
> > > 
> > > 
> > > It means that when we run stability tests, the case has a query, load, search latency increase phenomenon and datacoord CPU rise problem
> > 
> > 
> > I see.
> > @wangting0128 Do you think we can rerun this test on 2.2.3 branch and see if there's DataCoord CPU rise?
> 
> @elstic Please follow up on this question, thanks

@soothing-rain  Yes, we ran stability tests on the 2.2.3 branch during spring break and had problems with datacoord cpu and memory spikes.

**Comment by xiaofan-luan** [comment]
any progress?

**Comment by elstic** [comment]
Verified and fixed, datacoord cpu and memory usage is normal 
fix version:2.2.3
